### PR TITLE
chore: add examples, Go Workspaces

### PR DESCRIPTION
**Pull request by goferHiro**

### CODE REVIEW

File: go.work [hunk 1]
```diff
@@ -0,0 +1,5 @@
+go 1.19
```
Comment (bxcodec) [inline go.work:1]: What is this used for?

File: version.go [hunk 2]
```diff
@@ -0,0 +1,6 @@
+package gotcha
+
+// Version is the current release version.
+func Version() string {
+	return "1.0.0-beta.3"
```
Comment (bxcodec) [inline version.go:5]: So we need to update the version manually first before releasing?

File: go.work (hunk 1 above)
Comment (goferHiro) [reply go.work:1]: [this https://go.dev/blog/get-familiar-with-](https://go.dev/blog/get-familiar-with-workspaces)

Comment (goferHiro) [reply go.work:1]: Earlier implementation involved using a couple of replace directives but thats no longer needed with go workspaces

File: version.go (hunk 2 above)
Comment (goferHiro) [reply version.go:5]: no use scripts/auto.sh

Comment (goferHiro) [reply version.go:5]: or edit the tag in makefile and then make release

**Review by bxcodec — APPROVED** [review]

**Comment by bxcodec** [comment]
LGTM

**Pull request merged**